src/hotspot/share/nmt/nmtNativeCallStackStorage.hpp

```diff
@@ -41,19 +42,17 @@
 // We achieve this by using a closed hashtable for finding previously existing NCS:s and referring to them by an index that's smaller than a pointer.
 class NativeCallStackStorage : public CHeapObjBase {
 public:
-  struct StackIndex {
-    friend NativeCallStackStorage;
-    int32_t _stack_index;
-  public:
-    static constexpr const int32_t invalid = -1;
-    static bool equals(const StackIndex& a, const StackIndex& b) {
-      return a._stack_index == b._stack_index;
-    }
+  using StackIndex = int;
 
-    bool is_invalid() {
-      return _stack_index == invalid;
-    }
-  };
+  constexpr static const StackIndex invalid = std::numeric_limits<StackIndex>::max() - 1;
+
+  static bool equals(const StackIndex a, const StackIndex b) {
+    return a == b;
+  }
```

Comment:
Can get rid of, but is also fine imho.

Reply:
RegionData uses it in:

```
    static bool equals(const RegionData& a, const RegionData& b) {
      return a.flag == b.flag &&
             NativeCallStackStorage::StackIndex::equals(a.stack_idx, b.stack_idx);
    }
```

so I'd say it's nice to have it encapsulated like this.